### PR TITLE
First pass at a recommendation endpoint for user concepts

### PR DESCRIPTION
**Pull request by phamann**
First attempt at a recommendation endpoint \o/ to suggest concepts to a user based on the concepts they already follow. Uses collaborative filtering to achieve the result:
`(user:User)-[:followed]->(:Concept)<-[:followed]-(:User)-[rel:followed]->(other:Concept)` i.e. users whom follow the topics you follow also follow X.

- Add new `/recommendation/user/:id/concept` route
- Add new `user-concepts` controller and DB query
- Add new most popular poller to roll popular concepts from redis

### Current rules:
- Get collaborative filtered concepts
- filter out any with taxonomies `['genre','subjects', 'sections']`\
- filer out the top 20 most popular concepts
- order by amount of follows descending
- limit result set per query string.

### CODE REVIEW

**Comment by keirog** [comment]
Dat query tho

File: server/v2/lib/most-popular-concepts.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+class PopularConcepts {
+	constructor(opts) {
+		opts = opts || {};
+		this.data = [];
+		this.updateFrequency = opts.updateFrequency || '00 */1 * * *'; // every hour
+		this.job = new CronJob({
+			start: true,
+			runOnInit: true,
+			cronTime: this.updateFrequency,
+			onTick: () => { this.update(); }
```
Comment (laurieboyes) [inline server/v2/lib/most-popular-concepts.js]: Why the curly braces?

**Comment by laurieboyes** [comment]
Lovely :+1:

**Comment by keirog** [comment]
:+1:

**Pull request merged**